overlord/snapstate/handlers.go

```diff
@@ -1147,9 +1147,43 @@ func (m *SnapManager) doCopySnapData(t *state.Task, _ *tomb.Tomb) error {
 		}
 		return copyDataErr
 	}
+
+	// the migration hasn't been done - do it now
+	if opts.HideSnapDir && !opts.MigratedHidden {
+		if err := m.migrateToHiddenDir(t); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+func (m *SnapManager) migrateToHiddenDir(t *state.Task) error {
+	t.State().Lock()
+	snapsup, err := TaskSnapSetup(t)
+	t.State().Unlock()
+	if err != nil {
+		return err
+	}
+
+	// set the migrated status even on error, so we undo this later
```

Comment:
Can you please expand this comment a little, to explain what steps do we have to undo in this case?